cfme/infrastructure/pxe.py

```diff
@@ -642,68 +983,68 @@ def delete(self, cancel=True):
-        flash.assert_message_match('ISO Datastore "{}": Delete successful'.format(self.provider))
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove this ISO Datastore', handle_alert=not cancel)
+        main_view = self.create_view(PXEDatastoresView)
+        msg = 'ISO Datastore "{}": Delete successful'.format(self.provider)
+        main_view.flash.assert_success_message(msg)
 
     def refresh(self, wait=True, timeout=120):
         """ Refreshes the PXE relationships and waits for it to be updated
         """
-        navigate_to(self, 'Details')
-        last_time = pxe_details_page.last_refreshed.text
-        cfg_btn('Refresh Relationships', invokes_alert=True)
-        sel.handle_alert()
-        flash.assert_message_match(
-            'ISO Datastore "{}": Refresh Relationships successfully initiated'
-            .format(self.provider))
+        view = navigate_to(self, 'Details')
+        basic_info = view.entities.basic_information
+        last_time = basic_info.get_text_of('Last Refreshed On')
+        view.toolbar.configuration.item_select('Refresh Relationships', handle_alert=True)
+        view.flash.assert_success_message(('ISO Datastore "{}": Refresh Relationships successfully '
+                                           'initiated'.format(self.provider)))
         if wait:
-            wait_for(lambda lt: lt != pxe_details_page.last_refreshed.text,
-                     func_args=[last_time], fail_func=sel.refresh, num_sec=timeout,
+            wait_for(lambda lt: lt != basic_info.get_text_of('Last Refreshed On'),
+                     func_args=[last_time], fail_func=view.toolbar.reload.click, num_sec=timeout,
                      message="iso refresh")
 
     def set_iso_image_type(self, image_name, image_type):
         """
         Function to set the image type of a PXE image
         """
-        navigate_to(self, 'All')
-        iso_tree(self.provider, 'ISO Images', image_name)
-        cfg_btn('Edit this ISO Image')
-        fill(iso_image_type_form, {'image_type': image_type})
+        view = navigate_to(self, 'All')
+        view.sidebar.datastores.tree.click_path('All ISO Datastores', self.provider,
+                                                'ISO Images', image_name)
+        view.toolbar.configuration.item_select('Edit this ISO Image')
+        view.fill({'image_type': image_type})
         # Click save if enabled else click Cancel
-        try:
-            sel.click(form_buttons.save)
-        except:
-            sel.click(form_buttons.cancel)
+        if view.save.active:
+            view.save.click()
+        else:
+            view.cancel.click()
 
 
 @navigator.register(ISODatastore, 'All')
 class ISODatastoreAll(CFMENavigateStep):
+    VIEW = PXEDatastoresView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
         self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'PXE')
```

Comment:
Shouldn't this be replaced with a prerequisite of being in Infra/PXE before this step? Steps should ideally do one thing.